test/abstract_syntax_tree_location_data.coffee

```diff
@@ -0,0 +1,379 @@
+  eq node.loc.end.column, expected.loc.end.column, \
+    "Expected location end column #{reset}#{node.loc.end.column}#{red} to equal #{reset}#{expected.loc.end.column}#{red}"
+
+
+test "AST location data as expected for NumberLiteral node", ->
+  testAstLocationData '42',
+    start: 0
+    end: 2
+    range: [0, 2]
+    loc:
+      start:
+        line: 1
+        column: 0
+      end:
+        line: 1
+        column: 2
+
+test "AST location data as expected for InfinityLiteral node", ->
+  testAstLocationData 'Infinity',
+    start: 0
+    end: 8
+    range: [0, 8]
+    loc:
+      start:
+        line: 1
+        column: 0
+      end:
+        line: 1
+        column: 8
+
+test "AST location data as expected for NaNLiteral node", ->
+  testAstLocationData 'NaN',
+    start: 0
+    end: 3
+    range: [0, 3]
+    loc:
+      start:
+        line: 1
+        column: 0
+      end:
+        line: 1
+        column: 3
+
+test "AST location data as expected for IdentifierLiteral node", ->
+  testAstLocationData 'id',
+    start: 0
+    end: 2
+    range: [0, 2]
+    loc:
+      start:
+        line: 1
+        column: 0
+      end:
+        line: 1
+        column: 2
+
+test "AST location data as expected for StatementLiteral node", ->
+  testAstLocationData 'break',
+    start: 0
+    end: 5
+    range: [0, 5]
+    loc:
+      start:
+        line: 1
+        column: 0
+      end:
+        line: 1
+        column: 5
+
+  testAstLocationData 'continue',
+    start: 0
+    end: 8
+    range: [0, 8]
+    loc:
+      start:
+        line: 1
+        column: 0
+      end:
+        line: 1
+        column: 8
+
+  testAstLocationData 'debugger',
+    start: 0
+    end: 8
+    range: [0, 8]
+    loc:
+      start:
+        line: 1
+        column: 0
+      end:
+        line: 1
+        column: 8
+
+test "AST location data as expected for ThisLiteral node", ->
+  testAstLocationData 'this',
+    start: 0
+    end: 4
+    range: [0, 4]
+    loc:
+      start:
+        line: 1
+        column: 0
+      end:
+        line: 1
+        column: 4
+
+test "AST location data as expected for UndefinedLiteral node", ->
+  testAstLocationData 'undefined',
+    start: 0
+    end: 9
+    range: [0, 9]
+    loc:
+      start:
+        line: 1
+        column: 0
+      end:
+        line: 1
+        column: 9
+
+test "AST location data as expected for NullLiteral node", ->
+  testAstLocationData 'null',
+    start: 0
+    end: 4
+    range: [0, 4]
+    loc:
+      start:
+        line: 1
+        column: 0
+      end:
+        line: 1
+        column: 4
+
+test "AST location data as expected for BooleanLiteral node", ->
+  testAstLocationData 'true',
+    start: 0
+    end: 4
+    range: [0, 4]
+    loc:
+      start:
+        line: 1
+        column: 0
+      end:
+        line: 1
+        column: 4
+
+test "AST location data as expected for Access node", ->
+  testAstLocationData 'obj.prop',
```

Comment:
@GeoffreyBooth these tests currently aren't checking the nested nodes as desired because of your refactor of the shared AST testing helpers in helixbass#13